src/utils/userStatus.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * SPDX-FileCopyrightText: 2020 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import { t } from '@nextcloud/l10n'
+
+import type {
+	Conversation,
+	Participant,
+	ParticipantSearchResult,
+	ParticipantStatus,
+} from '../types/index.ts'
+
+/**
+ * Generate user status object to use as preloaded status with NcAvatar
+ *
+ * @param userData user data (from conversation, participant, search result)
+ */
+export function getPreloadedUserStatus(userData?: Conversation | Participant | ParticipantSearchResult): ParticipantStatus | undefined {
+	if (!userData || typeof userData !== 'object') {
+		return undefined
+	}
+
+	if ('statusMessage' in userData) {
+		// We preloaded the status when via participants API
+		return {
+			status: userData.status || null,
+			message: userData.statusMessage || null,
+			icon: userData.statusIcon || null,
+		}
+	}
+	if ('status' in userData && typeof userData.status === 'object') {
+		// We preloaded the status when via search API
+		return {
+			status: userData.status.status || null,
+			message: userData.status.message || null,
+			icon: userData.status.icon || null,
+		}
+	}
+	return undefined
+}
+
+/**
+ * Generate full status message for user according to its status data
+ *
+ * @param userData user data (formatted via getPreloadedUserStatus() )
+ */
+export function getStatusMessage(userData?: ParticipantStatus | ''): string {
```

Comment:
We are using this only with `getPreloadedUserStatus()`, it's better then to integrate them together from here.

Reply:
true, there was an issue before in TS part. Now all good